src/main/java/de/dennisguse/opentracks/ui/aggregatedStatistics/dailyStats/DailyPlottingModule.java

```diff
@@ -1,8 +1,45 @@
 package de.dennisguse.opentracks.ui.aggregatedStatistics.dailyStats;
+import com.github.mikephil.charting.data.Entry;
+import java.util.ArrayList; // import the ArrayList class
+import java.util.List; // import the List class
 
 /**
  * A module to aid in building the graph for visualizing daily statistics.
  */
 public class DailyPlottingModule {
 
+    /**
+     * Method that helps calculating moving average.
+     * @param entries This is a List of Entry objects that contains the original daily stats.
+     * @param frequency Integer representing the frequency at which moving average is sampled.
+     * @return A list containing Entry objects representing the moving averages.
+     */
+    public static List<Entry> getMovingAverage(List<Entry> entries, int frequency)
```

Comment:
The method doesn't need to be static because we're accessing it via an instance object